pages/blog/2020-04-06-flash-of-unstyled-dark-theme.md

```diff
@@ -0,0 +1,162 @@
+
+```javascript
+var themes = {
+  light: {
+    "color-fg": "#000",
+    "color-bg": "#fff",
+  },
+  dark: {
+    "color-fg": "#000",
+    "color-bg": "#fff",
+  }
+};
+```
+
+Now you can add this piece of *render-blocking* JavaScript in your `<head>`
+
+```javascript
+// Get root <html> node style interface
+var root = document.documentElement.style;
+var darkMatcher = window.matchMedia("(prefers-color-scheme: dark)");
+var prefersColorScheme = darkMatcher.matches ? "dark" : "light";
+
+window.__onSetTheme = function() {};
+window.__setTheme = function(themeName) {
+  var theme = themes[themeName];
+  Object.keys(theme).forEach(function(key) {
+    // Set global custom properties on root element
+    root.setProperty("--" + key, theme[key]);
+    window.__onSetTheme(theme)
+  });
+}
+window.__setTheme(prefersColorScheme);
+```
+
+This will set the color scheme for the initial render and also expose a global `__setTheme` function that you can call from within your React UI. You can also override the `__onSetTheme` to subscribe to when a theme change is applied.
+
+
+### Persist user theme preference
+
+The next thing you might want to do is to persist the theme that the user has opted for. A good candidate for this is to use `localStorage`. Update the `__setTheme` function
+
+```javascript
+window.__setTheme(themeName) {
+  window.localStorage.setItem("themeName", themeName);
+  ...
+}
+```
+
+Update the `setTheme` function call to attempt to use the persisted theme first, otherwise pass the `prefersColorScheme`
+
+```javascript
+var savedTheme = window.localStorage.getItem('themeName');
+setTheme(savedTheme || prefersColorScheme);
+```
+
+### Listen to thematic changes in user OS
+
+As a last added bonus, we can listen to if the user changes the preferred OS color scheme. It can be debatable if the page should should change if the user has explicitly chosen a theme to use. I can see this being more useful for apps that always should adapt to it's surroundings like a desktop or mobile application.
+
+```javascript
+darkmatcher.addListener(function() {
+  setTheme(darkMatcher.matches ? 'dark' : 'light');
+})
+```
+
+Note: If you're using a pure CSS media query solution as outlined in the first improved example, you already get this for free without any additional JavaScript.
+
+
+### Working examples
+
+For the sake of this article I've put together two working demos to show how everything fits together.
+
+- Next.js example: [Demo](https://next-flash-of-unstyled-dark-theme.netlify.com/) | [Source](https://github.com/danielstocks/flash-of-unstyled-dark-theme/tree/master/next)
+- Gatsby example: [Demo](https://gatsby-flash-of-unstyled-dark-theme.netlify.com/) | [Source](https://github.com/danielstocks/flash-of-unstyled-dark-theme/tree/master/gatsby)
+
+## Summary
```

Comment:
This is not a summary and it doesn't merit conclusions. If you can elaborate more so people have something to take home like a syntesis or a summary of steps like:

1 detect ... support
2 polyfils
3 global css
4 components
5 darkmatcher
... 

Other things haven't mention is colorisation of images https://medium.com/dev-channel/re-colorization-for-dark-mode-19e2e17b584b